mesh/mesh-io-tcpserver.c

```diff
@@ -314,51 +351,42 @@ static void tcpserver_io_init_done(void *user_data)
 	mesh_io->pvt->ready_callback(mesh_io->pvt->user_data, true);
 }
 
-static unsigned int tls_psk_server_cb(SSL *ssl, const char *identity,
-	unsigned char *psk, unsigned int max_psk_len)
+static bool match_tcpserver_conn_identity_buf(const void *a, const void *b)
 {
-	const char k1_info[] = { 'i', 'd', 'p', 's', 'k' };
-	unsigned int psk_len = 0;
-
-	struct mesh_io_private *pvt;
-	uint8_t net_id[8];
-	char s1_info[24];
-	uint8_t id[16];
-	char *id_str;
+	const struct tcpserver_conn	*conn = a;
+	const uint8_t			*identity = b;
 
-	pvt = SSL_get_ex_data(ssl, 0);
-
-	if (!pvt)
-		return 0;
+	return !memcmp(conn->identity_buf, identity, IDENTITY_BUF_LEN);
+}
 
-	// net_id = k3(net_key)
-	if (!mesh_crypto_k3(pvt->net_key, net_id))
+static unsigned int tls_psk_server_cb(SSL *ssl, const char *identity,
+	unsigned char *psk, unsigned int max_psk_len)
+{
+	uint8_t			identity_buf[IDENTITY_BUF_LEN];
+	struct mesh_io		*mesh_io;
+	struct silvair_io	*silvair_io;
+	struct tcpserver_conn	*conn;
+
+	if (!str2hex(identity, strlen(identity), identity_buf,
+						     sizeof(identity_buf))) {
+		l_error("Could not convert identity str to hex.");
 		return 0;
+	}
 
-	// s1_info = (uuid|net_id)
-	memcpy(s1_info, pvt->uuid, UUID_LEN);
-	memcpy(s1_info + UUID_LEN, net_id, sizeof(net_id));
+	mesh_io		= SSL_get_ex_data(ssl, 0);
+	silvair_io	= SSL_get_ex_data(ssl, 1);
 
-	// id = s1(s1_info)
-	if (!mesh_crypto_s1(s1_info, sizeof(s1_info), id))
+	conn = l_queue_find(mesh_io->pvt->conns,
+			match_tcpserver_conn_identity_buf, identity_buf);
+	if (!conn)
 		return 0;
 
-	id_str = l_util_hexstring_upper(id, sizeof(id));
-
-	// compare str format 'identity' and 'id_str'
-	if (!strcmp(identity, id_str))
-		goto done;
-
-	// psk = k1(dev_key, id, "idpsk")
-	if (!mesh_crypto_k1(pvt->dev_key, id, k1_info, sizeof(k1_info), psk))
-		goto done;
+	conn->io = silvair_io;
 
-	// psk is constant length
-	psk_len = 16;
+	if (conn->conn_stat)
+		conn->conn_stat->connected = true;
```

Comment:
Same: emit `PropertiesChanged` here.

Reply:
conn_stat will provide functions that will emit signals (as we discussed). I've put it here just for test purpose.